pkg/gui/types/keybindings.go

```diff
@@ -25,6 +25,13 @@ type Binding struct {
 
 	// to be displayed if the keybinding is highlighted from within a menu
 	Tooltip string
+
+	// Function to decide whether the command is enabled. If this returns an
+	// empty string, it is; if it returns a non-empty string, it is disabled and
+	// we show the given text in an error message when trying to invoke it.
+	// When left nil, the command is always enabled.
+	// Note that this function must not do expensive calls.
+	Enabled func() string
```

Comment:
It's hard to know what this function signature represents just from the name. How does `GetDisabledReason` sound to you?

Reply:
That's true, but I was hoping that explaining this in the comment would be enough. I kind of like the current pattern of having two handlers, e.g. `rebase` and `rebaseEnabled`; this wouldn't work so well then. `rebase` and `getDisabledReasonForRebase`? Hm, could do that I suppose.

But I don't have a very strong opinion, happy to accept your decision.

Reply:
It's more verbose indeed but I think it minimises surprise

Reply:
OK, changed in 5bc7188d6b1a. The changes to the client code couldn't usefully be made as fixups, so I squashed them in. I changed the names of some of the helper functions in `local_commits_controller.go`, it's worth looking over them again.